packages/jupyter-ai/src/components/chat-messages.tsx

```diff
@@ -129,6 +150,7 @@ export function ChatMessages(props: ChatMessagesProps): JSX.Element {
     }
   }, [props.messages]);
 
+  const sortedMessages = sortMessages(props.messages);
```

Comment:
Can you store `sortedMessages` in a new state variable, and only set this state variable when `props.messages` is changed? This ensures that sorting is only performed as-necessary. You can do this via an effect hook:

```ts
const [sortedMessages, setSortedMessages] = useState<AiService.ChatMessage[]>([])

useEffect(() => {
  ... // call sortMessages()
}, [props.messages])
```